src/dlite-type.c

```diff
@@ -1198,23 +1204,50 @@ int dlite_type_scan(const char *src, int len, void *p, DLiteType dtype,
         return errx(dliteParseError, "relation should have 3 (optionally 4) elements");
       m = tokens->end - tokens->start;
       if (tokens->type == JSMN_ARRAY) {
+        size_t bufsize=0;
+        char *buf=NULL;
         if (!(t = jsmn_element(src, tokens, 0))) return -1;
-        rel->s = strndup(src + t->start, t->end - t->start);
+        if (strnput_unquote(&buf, &bufsize, 0, src + t->start,
```

Comment:
seems to be a lot of hard-to-read copy-pasted code here. Please create a function, or even a macro to simplify the reading.

Reply:
Good point. Done